### PR TITLE
fix!: drop `watchChange` hook for esbuild

### PR DESCRIPTION
**Pull request by sxzz**

### CODE REVIEW

**Comment by antfu** [comment]
Reasoning: 

`esbuild` reworked the watch API in v0.17.0: https://github.com/evanw/esbuild/releases/tag/v0.17.0, which makes unplugin not able to emulate watch behavior (namely: https://github.com/evanw/esbuild/issues/2823). As the previous watch feature is also a hacky workaround and not working perfectly, we decided to drop the support for `watchChange` and make it clear.

File: README.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ Currently supports:
 | `transformInclude`<sup>2</sup> | ✅ | ✅ | ✅ | ✅ | ✅ |
 | [`transform`](https://rollupjs.org/guide/en/#transformers) | ✅ | ✅ | ✅ | ✅ | ✅ <sup>3</sup> |
 | [`watchChange`](https://rollupjs.org/guide/en/#watchchange) | ✅ | ✅ | ✅ | ✅ | ✅ |
-| [`buildEnd`](https://rollupjs.org/guide/en/#buildend) | ✅ | ✅ | ✅ | ✅ | ✅ |
```
Comment (lforst) [inline README.md:31]: Hi, is removing build end support here intentional?

Comment (sxzz) [reply README.md:31]: Oh, it was a mistake.

Comment (sxzz) [reply README.md:31]: fixed

Comment (lforst) [reply README.md:31]: Awesome, thank you!

**Pull request merged**